### ISSUE TITLE
Failure setting indexers as search peers

### ISSUE THREAD

**Issue by jdso7** [splunk/docker-splunk#48]
Hi, I'm getting the following error which causes the search head container to stop.

TASK [splunk_search_head : Set all indexers as search peers] *******************
Saturday 27 October 2018  22:04:52 +0000 (0:00:03.579)       0:02:47.351 ****** 
fatal: [localhost]: FAILED! => {"msg": "The conditional check 'task_result.rc == 0' failed. The error was: error while evaluating conditional (task_result.rc == 0): 'task_result' is undefined"}

In my scenario I have 2 indexers and 1 search head. I've used 2idx2sh.yaml as a starting point for my file. I've attached my yaml file. 

[docker-compose.yml.txt](https://github.com/splunk/docker-splunk/files/2522259/docker-compose.yml.txt)

Thanks

**Comment by matthewmodestino** [comment]
Hi jdso7! Have you followed the steps for using docker run to create a `defaults.yaml`? 

https://github.com/splunk/docker-splunk/blob/master/documentation/ADVANCED.md#defaultyml-valid-options

If you check out the advanced documentation section, or the kubernetes examples (you do not need to know.use k8s for this to be relevant): 
https://github.com/splunk/docker-splunk/tree/master/test_scenarios/kubernetes, there is a walkthrough of creating this file to serve the base configs to any container in a deploy, and serving it centrally using nginx. You could also just ensure to copy it into the container when you build it, etc. 

This is an inportant file to get right to ensure the options you want/need (ie. clustering, pass4SymKey, replication ports, etc)

My hunch is we just need to get your `defaults.yaml` sorted to match the desired environment you want to build. Do you want the indexers to be clustered, or just standalone, etc. 

Try using docker run to generate one, and we can start there to ensure all the items you want globally are set and accounted for in `defaults.yaml` or in env variables in the container. 

I think the last time I saw this error, it was because there was no Master Node and Indexer Clustering was enabled.

**Comment by jdso7** [comment]
Hi matthewmodestino, 

Yes, I have been looking at that documentation. I've used environment variables instead of passing values in the defaults.yaml file.

It's a bit hard to understand the error from the log, even in debug. 
So thank you for your hints, I'll double-check those settings.

**Comment by arctan5x** [comment]
Hey @JasonConger,

I have investigated this issue and made a fix for it in the Splunk-Ansible Repo.

<img width="1409" alt="screen shot 2018-11-05 at 3 28 33 pm" src="https://user-images.githubusercontent.com/6732195/48033295-a2fc4180-e10f-11e8-8142-6d3ade998370.png">

Seems like peering works after the fix. Note that I took out volumes and license configurations from your yaml since it is your custom resource.
